### DISCUSSION TITLE
Filter table not applied on `diesel migration --diff-schema`

### DISCUSSION THREAD

**Discussion by forest1102** [diesel-rs/diesel#3712] [Ideas]
I am trying out a new option `diff-schema`.  
However, when I use the `filter` field in diesel.toml just omits tables used in other crates. `diesel migration generate edit_column_name --diff-schema` results in writing code to delete tables not listed in the filter fields. I expect migration do nothing about unlisted tables.

# Inputs

initial database info
```sql
CREATE TABLE "table_a"(
	"id" INT4 NOT NULL PRIMARY KEY,
	"code" VARCHAR NOT NULL
);
CREATE TABLE "table_b"(
	"id" INT4 NOT NULL PRIMARY KEY,
);
CREATE TABLE "table_c"(
	"id" INT4 NOT NULL PRIMARY KEY,
);
```

diesel.toml
```  :diesel.toml
[print_schema]



file = "schema_a/src/schema.rs"
filter = { only_tables = [
    "^table_a$",
] }
```

schema_a/src/schema.rs
``` diff
diesel::table! {
    table_a (id) {
        id -> Int8,
-       code -> Varchar,
+      script -> Varchar,
    }
}
```

command
```shell
diesel migration generate edit_column_name --diff-schema
```

## Actual Result

up.sql
```sql
DROP TABLE IF EXISTS "table_b";
ALTER TABLE "table_a" DROP COLUMN "code";
ALTER TABLE "table_a" ADD COLUMN "script" VARCHAR NOT NULL;
DROP TABLE IF EXISTS "table_c";
```

## Expected

up.sql
```sql
ALTER TABLE "table_a" DROP COLUMN "code";
ALTER TABLE "table_a" ADD COLUMN "script" VARCHAR NOT NULL;
```

## Version
```
diesel 
 Version: 2.1.0
 Supported Backends: postgres
```

**Comment by weiznich** [comment]
Thanks for writing this post.

Can you please provide a self contained reproducible example for this? That needs to include at least the following information:

* Initial schema (for example via migrations)
* Your modified `schema.rs` file
* An exact list of commands you've run
* Information about your database backend, diesel version, ...
* An example of what you would like to expect the migration to contain instead.

**Reply by forest1102** [reply]
Sorry for lack of information. 
I added files I used and database definition.

**Reply by weiznich** [reply]
Thanks for the update. This is an issue in diesel. I've opened https://github.com/diesel-rs/diesel/issues/3717 to track this issue.
